dockstore-client/src/main/java/io/dockstore/client/cli/nested/WesLauncher.java

```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    /**
+     * Obtains the workflow on Dockstore
+     *
+     * @param workflowClient The WorkflowClient
+     * @param workflowEntry The workflow entry (i.e. github.com/myRepo/myWorkflow:version)
+     * @return A Workflow object for the corresponding workflow+version
+     */
+    public static Workflow getWorkflowForEntry(WorkflowClient workflowClient, String workflowEntry) {
+        String[] parts = workflowEntry.split(":");
+        String path = parts[0];
+        String version = workflowClient.getVersionID(workflowEntry);
+        return workflowClient.getWorkflowsApi().getPublishedWorkflowByPath(path, null, null, version);
+    }
+
+    /**
+     * Creates a TRS url for a Dockstore entry
+     *
+     * @param workflowClient The WorkflowClient
+     * @param workflowEntry The workflow entry (i.e. github.com/myRepo/myWorkflow:version)
+     * @param workflow The workflow object we are creating a TRS url for
+     * @return A string representing a TRS URL for an entry
+     */
+    public static String combineTrsUrlComponents(WorkflowClient workflowClient, String workflowEntry, Workflow workflow) {
+        ApiClient client = workflowClient.getClient().getGa4Ghv20Api().getApiClient();
+
+        // Entries are passed in the form {PATH}:{VERSION} or {PATH}
+        final String[] pathAndVersion = workflowEntry.split(":");
+        final String path = pathAndVersion[0];
+
+        // Calculate the values needed to supply a TRS URL
+        final String basePath = workflowClient.getClient().getGa4Ghv20Api().getApiClient().getBasePath();
+        final String versionId = workflowClient.getVersionID(workflowEntry);
+        final String entryId = workflowClient.getTrsId(path);
+        final String type = "PLAIN_" + workflow.getDescriptorType().getValue();
+
+        // Escape each of the URL path values
+        final String escapedId = client.escapeString(entryId);
+        final String escapedVersionId = client.escapeString(versionId);
+        final String escapedType = client.escapeString(type);
+        final String escapedRelativePath = client.escapeString(workflow.getWorkflowPath());
+
+        // Return the TRS URL for a given entry
+        return MessageFormat.format("{0}/ga4gh/trs/v2/tools/{1}/versions/{2}/{3}/descriptor/{4}",
+            basePath,
+            escapedId,
+            escapedVersionId,
+            escapedType,
+            escapedRelativePath);
+    }
+
+    /**
+     * Attempts to load a file from the provided path. Errors out if the file doesn't exist.
+     *
+     * @param filePath Path to a file or null
+     * @return An Optional File object, this may be empty if the filepath is null
+     */
+    public static Optional<File> fetchFile(String filePath) {
+        if (filePath == null) {
+            return Optional.empty();
+        }
+
+        // Verify the file path exists
+        Path path = Paths.get(filePath);
+        if (!Files.exists(path)) {
```

Comment:
I think, but you should verify, that  `Files.exists` return true if it's a directory (I know that's true for `File.exists`). If yes, do you want this?

Reply:
Good catch, `File.exists` returned true for both files and directories, which caused a NPE down the line. 

Updated to `File.isRegularFile` which explicitly checks if it is a file, also added a test to check the directory case: https://github.com/dockstore/dockstore-cli/pull/100/commits/b6b66faa279194b7955db8111f745ef4cc623df5